src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHostBuilderExtensions.cs

```diff
@@ -27,27 +27,63 @@ public static class SystemdHostBuilderExtensions
         ///     notifications. See https://www.freedesktop.org/software/systemd/man/systemd.service.html.
         ///   </para>
         /// </remarks>
-        /// <param name="hostBuilder">The <see cref="IHostBuilder"/> to use.</param>
-        /// <returns></returns>
+        /// <param name="hostBuilder">The <see cref="IHostBuilder"/> to configure.</param>
+        /// <returns>The <paramref name="hostBuilder"/> instance for chaining.</returns>
         public static IHostBuilder UseSystemd(this IHostBuilder hostBuilder)
         {
             if (SystemdHelpers.IsSystemdService())
             {
                 hostBuilder.ConfigureServices((hostContext, services) =>
                 {
-                    services.Configure<ConsoleLoggerOptions>(options =>
-                    {
-                        options.FormatterName = ConsoleFormatterNames.Systemd;
-                    });
-
-                    // IsSystemdService() will never return true for android/browser/iOS/tvOS
-#pragma warning disable CA1416 // Validate platform compatibility
-                    services.AddSingleton<ISystemdNotifier, SystemdNotifier>();
-                    services.AddSingleton<IHostLifetime, SystemdLifetime>();
-#pragma warning restore CA1416 // Validate platform compatibility
+                    AddSystemdLifetime(services);
                 });
             }
             return hostBuilder;
         }
+
+        /// <summary>
+        /// Configures the lifetime of the <see cref="IHost"/> built from <paramref name="services"/> to
+        /// <see cref="SystemdLifetime"/> provides notification messages for application started
+        /// and stopping, and configures console logging to the systemd format.
+        /// </summary>
+        /// <remarks>
+        ///   <para>
+        ///     This is context aware and will only activate if it detects the process is running
+        ///     as a systemd Service.
+        ///   </para>
+        ///   <para>
+        ///     The systemd service file must be configured with <c>Type=notify</c> to enable
+        ///     notifications. See https://www.freedesktop.org/software/systemd/man/systemd.service.html.
```

Comment:
Would be good to do the same as https://github.com/dotnet/aspnetcore/pull/41254

```suggestion
        ///     notifications. See <see href="https://www.freedesktop.org/software/systemd/man/systemd.service.html"/>.
```